zetaclient/zetacore/tx_test.go

```diff
@@ -285,6 +288,63 @@ func TestZetacore_MonitorVoteInboundResult(t *testing.T) {
 
 		require.NoError(t, err)
 	})
+
+	t.Run("monitor inbound vote out of gas", func(t *testing.T) {
+		// ARRANGE
+		msg := &crosschaintypes.MsgVoteInbound{}
+
+		// given mock tx result
+		cometbftMock.SetTxResult(&coretypes.ResultTx{
+			TxResult: abcitypes.ExecTxResult{
+				Log:       "out of gas",
+				GasWanted: 1000,
+			},
+		})
+
+		// should not adjust the gas limit, no out-of-gas failure so far
+		gasLimit, retryGasLimit, retryable := client.getAdjustedVoteInboundGasLimit(msg.Digest(), 1000, 2000)
+		require.Equal(t, uint64(1000), gasLimit)
+		require.Equal(t, uint64(2000), retryGasLimit)
+		require.True(t, retryable)
+
+		// ACT 1
+		// monitor out of gas error, record it
+		err := client.MonitorVoteInboundResult(ctx, sampleHash, 0, msg, nil)
+		require.NoError(t, err)
+
+		// ASSERT 1.1
+		// should adjust the gas limit
+		gasLimit, retryGasLimit, retryable = client.getAdjustedVoteInboundGasLimit(msg.Digest(), 1000, 2000)
+		require.Equal(t, uint64(2000), gasLimit)
+		require.Zero(t, retryGasLimit)
+		require.True(t, retryable)
+
+		// ASSERT 1.2
+		// not retryable, given gas limits are too low
+		gasLimit, retryGasLimit, retryable = client.getAdjustedVoteInboundGasLimit(msg.Digest(), 500, 1000)
+		require.Zero(t, gasLimit)
+		require.Zero(t, retryGasLimit)
+		require.False(t, retryable)
+
+		// monitor it again with same 'GasWanted', nothing new to record
+		err = client.MonitorVoteInboundResult(ctx, sampleHash, 0, msg, nil)
```

Comment:
Looks to be legit comments?

Reply:
https://github.com/zeta-chain/node/pull/4291/commits/274eafa116b2830ae37c6da665c07e5149dd0364